keymanagementrequirements.md

```diff
@@ -0,0 +1,79 @@
+            	{
+            		"trustedRoots": [{
+            			"root": "test.crt"
+            		}]
+            	}
+            ],
+            "trustedArtifacts": []
+        }
+    }
+```
+- Trust Store: The trust store defines the relationship between signing keys and artifacts that are used at validation time to determine whether to trust an artifact with a crytptographically valid signature. The trust store will relate a scope (any source, specific registry, specific repository, or specific target) with a certificate (for root key, intermediate, or signing key) or key repository (for automated key distribtuion). It is not recommended to use a signing key as this will cause signature validation to fail if the signing key is rotated. An example trustStore where the first root is trusted for artifacts from any registry and the second root is only trusted for artifacts from "registry.wabbit-networks.io" :
+```
+    {
+    	"trustedRoots": [{
+ 	    		"root": "-----BEGIN CERTIFICATE-----examplecertificate-----END CERTIFICATE-----"
+ 	    	},
+ 	    	{
+ 	    		"root": "-----BEGIN CERTIFICATE-----examplecertificate-----END CERTIFICATE-----",
+ 	    		"scope": "registry.wabbit-networks.io"
+ 	    	}
+ 	    ]
+    }
+```
+- Trust Store Key Information: The key information configured in the trust store will be cryptographically verifiable and contain:
+    - Public Key: Will be used to verify signed artifacts.
+    - Signed Identity: Identity of creator of the signing key.
+    - (Optional) Issuer: If the key is an intermediate this will describe the owner of the root.
+    - (Optional) Mechanism to check whether signatures generated with the signing key have been rescinded.
+
+## Requirements
+- Signing an artifact MUST NOT require the publisher to perform additional actions with a registry/repository or registry operator. Uploading a signature MAY require additional actions.
+- Retrieving a signature MUST NOT require the deployer to perform additional actions with a registry/repository or registry operator beyond those required to pull an unsigned artifact.
+- Artifact integrity, source, and signature expiry MUST be verifiable from the signature AND NOT require additional calls to the registry/repository.
+- Signature allow list/deny list MAY require additional calls that will be defined in a separate document.
+- Copying an artifact from one repository to another SHOULD NOT invalidate the signature on the artifact.
+- Publishers SHOULD be able to sign with keys stored on their local machines, secure tokens, Hardware Security Modules (HSMs), or cloud based Key Management Services.
+- Publishers SHOULD be able to generate multiple signatures for a single artifact.
+- Publisher admins MUST have a mechanism to revoke signatures to indicate they are no longer trusted. 
+- Trust stores MUST be configurable by the deployer admin.
+- Deployer admins MUST be able to configure trusted entities for individual repositories and targets.
```

Comment:
I'd add something along the lines that "Deployer MUST be able to trust a signing key with a certificate chain that leads to a trusted root or intermediate certificate according to the trust policy". We're still missing the detail of how to communicate that certificate chain (e.g. does it get pushed to the registry with the signing artifact, along side the signature, or from an external discovery system).